tests/integration/mounts_sshfs.bats

```diff
@@ -19,95 +18,287 @@ function teardown() {
+	run grep "\bnoexec\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	# Unspecified flags must be cleared.
+	run grep "\bnosuid\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	run grep "\bnodev\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	run grep "\bnoatime\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	# runc produces the kernel relatime default.
+	run grep "\brelatime\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+
+	# Now try with a user namespace.
+	update_config ' .linux.namespaces += [{"type": "user"}]
+		| .linux.uidMappings += [{"hostID": 100000, "containerID": 0, "size": 65534}]
+		| .linux.gidMappings += [{"hostID": 100000, "containerID": 0, "size": 65534}] '
+
+	# Setting flags that are not locked should work.
+	pass_sshfs_bind_flags "nodev,nosuid,noatime" "bind,ro,exec"
+	# The flagset should be the union of the two.
+	run grep "\bro\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+	run grep "\bnoexec\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	# (Unspecified MNT_LOCKED flags are inherited.)
+	run grep "\bnosuid\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+	run grep "\bnodev\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+	run grep "\bnoatime\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+}
+
+@test "runc run [bind mount {no,rel,strict}atime semantics]" {
+	requires root
+
+	function is_strictatime() {
+		# There is no "strictatime" in /proc/self/mounts.
+		run grep "\bnoatime\b" <<<"${1:-$mnt_flags}" && [ "$status" -ne 0 ]
+		run grep "\brelatime\b" <<<"${1:-$mnt_flags}" && [ "$status" -ne 0 ]
+		run grep "\bnodiratime\b" <<<"${1:-$mnt_flags}" && [ "$status" -ne 0 ]
+	}
+
+	# In contrast to mount(8), runc will clear atime flags and set them to the
+	# kernel default (relatime) ...
+	pass_sshfs_bind_flags "noatime" "bind"
+	# runc produces the kernel relatime default.
+	run grep "\bnoatime\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	run grep "\brelatime\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+
+	# ... or a sane default (strictatime) if relatime was explicitly disabled
+	# with "norelatime".
+	pass_sshfs_bind_flags "noatime" "bind,norelatime"
+	is_strictatime "$mnt_flags"
+
+	# Setting a flag that mount(8) would combine should result in only the
+	# requested flag being set.
+	pass_sshfs_bind_flags "noatime" "bind,nodiratime"
+	run grep "\bnoatime\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	run grep "\bnodiratime\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+	# The kernel applies the default relatime setting for nodiratime if no
+	# other atime setting is requested, and runc mirrors that behaviour.
+	run grep "\brelatime\b" <<<"$mnt_flags" && [ "$status" -eq 0 ]
+
+	# Clearing flags that mount(8) would not clear works.
+	pass_sshfs_bind_flags "nodiratime" "bind,strictatime"
+	# There is no "strictatime" in /proc/self/mounts.
+	run grep "\brelatime\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	run grep "\bnodiratime\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
+	run grep "\bnoatime\b" <<<"$mnt_flags" && [ "$status" -ne 0 ]
```

Comment:
Isn't it the same as `is_strictatime "$mnt_flags"`?

Reply:
Ah, I forgot to replace this one.